homeassistant/components/overkiz/coordinator.py

```diff
@@ -101,11 +104,23 @@ async def _async_update_data(self) -> dict[str, Device]:
             if event_handler := EVENT_HANDLERS.get(event.name):
                 await event_handler(self, event)
 
+        # Reset update interval to previous value
+        # when all executions are finished
         if not self.executions:
-            self.update_interval = UPDATE_INTERVAL
+            self._restore_update_interval()
 
         return self.devices
 
+    def set_update_interval(self, update_interval: timedelta) -> None:
```

Comment:
Is this intentionally overwriting `original_update_interval` again? I was thinking about thie case where this function is called twice, then `_restore_update_interval` doesn't restore the original value anymore, it restores an updated original value.

Perhaps you could walk me through what is supposed to be happening in a little more detail if i'm mis-undertanding the cases since it seems like there may be multiple interesting cases being exercised here, but hard to confirm without tests exercising the issues.

Reply:
Do you have an example of data update coordinator tests? I looked into that a while ago, but I was not able to get started 🙁.

The other comments I will have a look at tomorrow.

Reply:
The tests for using an update coordinator can look like any integration. For example, when [updating an integration](https://github.com/home-assistant/core/pull/74690/files) to use a coordinator I didn't need to change the tests really. But you can use google calendar as an example...

Reply:
I added an extra check in 442019d3812c1f3c62ebe72bfaa0e84868b1e635 to only update the original interval if the new one is different. In theory this should not happen, but looking at the code and all events, it could happen when Somfy does not broadcast the events in order.

Reply:
Should it just be if `self.original_update_interval is None` so that it's only possible to be set when its not already set?

Reply:
I'm still having a little trouble following since it seems like `original_update_interval` should just never be updated once its set, given the name "original". 

Given original is set in the constructor, can this function only update `self.update_interval = update_interval`?